### PR TITLE
FEATURE: Add unit tests for server/plugins part-2

### PR DESCRIPTION
**Pull request by caohhung**

### CODE REVIEW

File: tests/server/test_plugins.py [hunk 1]
```diff
@@ -4,7 +4,12 @@
 
 import copy
 import datetime
+import lockfile
```
Comment (pbenas) [inline tests/server/test_plugins.py]: please add the `lockfile` requirement to the Radme's testing section

File: tests/server/test_plugins.py [hunk 2]
```diff
@@ -486,3 +491,464 @@ def test_history_of_collect_new_result(self):
...
+        assert 'status' in worker.result and worker.result['status'] == 'OK'
+        assert 'info' in worker.result['messages']
+        assert worker.result['messages']['info'] == [os.uname()[1]]
+
+    def test_running_worker_process_title_should_be_changed(self):
+        expected = 'smokerd plugin Hostname'
+        worker = server_plugins.PluginWorker(**self.conf_worker)
+        worker.run()
+        procs = get_process_list()
+        assert expected in procs.values()
```
Comment (pbenas) [inline tests/server/test_plugins.py]: Fails on my machine... maybe a race condition? 
```bash
___________________________________ TestPluginWorker.test_running_worker_process_title_should_be_changed ____________________________________

self = <tests.server.test_plugins.TestPluginWorker object at 0x102148210>

    def test_running_worker_process_title_should_be_changed(self):
        expected = 'smokerd plugin Hostname'
        worker = server_plugins.PluginWorker(**self.conf_worker)
        worker.run()
        procs = get_process_list()
>       assert expected in procs.values()
E       assert 'smokerd plugin Hostname' in ['kernel_task', 'launchd', 'storelegacy', 'ViewBridgeAuxiliary', 'KeePassX', 'periodic-wrapper', ...]
E        +  where ['kernel_task', 'launchd', 'storelegacy', 'ViewBridgeAuxiliary', 'KeePassX', 'periodic-wrapper', ...] = <built-in method values of dict object at 0x10212f6e0>()
E        +    where <built-in method values of dict object at 0x10212f6e0> = {0: 'kernel_task', 1: 'launchd', 45: 'syslogd', 46: 'UserEventAgent', ...}.values

tests/server/test_plugins.py:534: AssertionError
```

File: tests/server/test_plugins.py [hunk 3]
```diff
@@ -486,3 +491,464 @@ def test_history_of_collect_new_result(self):
...
+        test_params = {
+            'Command': 'InvalidCommand'
+        }
+        params = dict(self.params_default, **test_params)
+        worker = server_plugins.PluginWorker(name='InvalidCommand',
+                                             queue=self.queue, params=params)
+        worker.run()
+        assert 'status' in worker.result and worker.result['status'] == 'ERROR'
+        assert 'warn' in worker.result['messages']
+        assert worker.result['messages']['error'] == expected_message
```
Comment (pbenas) [inline tests/server/test_plugins.py]: Fails on my machine... Probably related to bash version or bash configuration. Changing to regexp could probably solve this. 
```bash
_________________________________________________ TestPluginWorker.test_run_invalid_command _________________________________________________

self = <tests.server.test_plugins.TestPluginWorker object at 0x10216f110>

    def test_run_invalid_command(self):
        expected_message = ['/bin/sh: 1: InvalidCommand: not found']
        test_params = {
            'Command': 'InvalidCommand'
        }
        params = dict(self.params_default, **test_params)
        worker = server_plugins.PluginWorker(name='InvalidCommand',
                                             queue=self.queue, params=params)
        worker.run()
        assert 'status' in worker.result and worker.result['status'] == 'ERROR'
        assert 'warn' in worker.result['messages']
>       assert worker.result['messages']['error'] == expected_message
E       assert ['/bin/sh: In...nd not found'] == ['/bin/sh: 1: ...d: not found']
E         At index 0 diff: '/bin/sh: InvalidCommand: command not found' != '/bin/sh: 1: InvalidCommand: not found'
E         Use -v to get the full diff
```

**Comment by pbenas** [comment]
Good job Hung! PEP8 compliant for the first attempt, you've made a lot of progress since your previous pull request. :)

File: tests/server/test_plugins.py [hunk 4]
```diff
@@ -486,3 +491,464 @@ def test_history_of_collect_new_result(self):
...
+        assert result.validated
+
+
+def get_process_list():
+    procs = dict()
+    for proc in psutil.process_iter():
+        try:
+            pinfo = proc.as_dict(attrs=['pid', 'cmdline'])
+            procs[pinfo['pid']] = pinfo['cmdline'][0]
+        except (psutil.NoSuchProcess, IndexError, TypeError):
```
Comment (caohhung) [inline tests/server/test_plugins.py]: if 'cmdline' is Null, psutil will return :
 + 'None' on Mac
 + An emty list on linux

**Pull request merged**